matchzoo/preprocessor/arci_preprocessor.py

```
@@ -0,0 +1,170 @@
+        ]
+
+    def fit(self, inputs: typing.List[tuple]):
+        """
+        Fit pre-processing context for transformation.
+
+        :param inputs: Inputs to be preprocessed.
+        :return: class:`ArcIPreprocessor` instance.
+        """
+        vocab = []
+        units = self._prepare_stateless_units()
+
+        logger.info("Start building vocabulary & fitting parameters.")
+
+        # Convert user input into a datapack object.
+        self._datapack = self.segmentation(inputs, stage='train')
+
+        # Loop through user input to generate words.
+        # 1. Used for build vocabulary of words (get dimension).
+        # 2. Cached words can be further used to perform input
+        #    transformation.
+        left = self._datapack.left
+        right = self._datapack.right
+
+        for idx, row in tqdm(left.iterrows()):
+            # For each piece of text, apply process unit sequentially.
+            text = row.text_left
+            for unit in units:
+                text = unit.transform(text)
+            vocab.extend(text)
+            # cache words for transformation.
+            self._cache_left.append((row.name, text))
+
+        for idx, row in tqdm(right.iterrows()):
+            # For each piece of text, apply process unit sequentially.
+            text = row.text_right
+            for unit in units:
+                text = unit.transform(text)
+            vocab.extend(text)
+            # cache words for transformation.
+            self._cache_right.append((row.name, text))
+
+        # Initialize a vocabulary process unit to build words vocab.
+        self._vocab_unit.fit(vocab)
+
+        if os.path.isfile(self._embedding_file):
+            embed_module = Embedding(embedding_file=self._embedding_file)
+            embed_module.build(self._vocab_unit.state['term_index'])
+            self._context['embedding_mat'] = embed_module.embedding_mat
+        else:
+            logger.info("Embedding file [{}] not found."
```

when can not find embedding file, we should raise an Exception.